src/Client.cpp

```diff
@@ -996,3 +1006,32 @@ void CClient::ParseIdentifier(const CString& sAuthLine) {
 		m_sUser = sAuthLine;
 	}
 }
+
+void CClient::SetNetworkCaps(const SCString& ssCaps)
+{
+	if (m_ssNetworkCaps == ssCaps) {
+		return;
+	}
+
+	SCString ssDelCaps;
+	SCString ssNewCaps;
+
+	if (!m_ssNetworkCaps.count("away-notify") && ssCaps.count("away-notify")) {
```

Comment:
If you're talking about this exact logic, you can try something with having an array of 2 sets, and counting an index in that array; but it'll be more confusing than this.